### PR TITLE
[shelly] Revise fix for Gen1 initialization problem for manually created Things

### PR DESCRIPTION
**Pull request by markus7017**
This PR replaces #17009 with a more solid solution supporting gen 1/2/3 devices discovered by UI for defined in a .things file

### CODE REVIEW

**Comment by jlaur** [comment]
Related to https://github.com/openhab/openhab-addons/pull/17009/files#r1667631650

Since I proposed this change, I would appreciate another @openhab/add-ons-maintainers to have a look.

**Review by jlaur — APPROVED** [review]
LGTM

**Review by jlaur — CHANGES_REQUESTED** [review]
Please see https://github.com/openhab/openhab-addons/issues/17013#issuecomment-2212487299

**Comment by markus7017** [comment]
- README updated
- A bug in Gen3 init resulting in isGen2 = false
- shellywalldisplay added to isGeneration2()

File: bundles/org.openhab.binding.shelly/README.md [hunk 1]
```diff
@@ -1551,6 +1560,14 @@ See notes on discovery of Shelly BLU devices above.
...
 |         | lowBattery    | Switch   | yes       | Low battery alert (< 20%)                               |
 
+## Shelly BLU Gateway (thing-type: shellyblugw)
+
+| Group   | Channel       | Type     | read-only | Description                                             |
+| ------- | ------------- | -------- | --------- | ------------------------------------------------------- |
+
+There no no additional channels beside the device group.
+
+
```
Comment (jlaur) [inline bundles/org.openhab.binding.shelly/README.md:1570]: I guess there is no reason to include an empty table?
```suggestion
There are no additional channels beside the device group.

```

File: bundles/org.openhab.binding.shelly/README.md [hunk 2]
```diff
@@ -126,6 +127,14 @@ The binding provides the same feature set across all devices as good as possible
 | shellyblumotion   | Shelly BLU Motion                                      | SBMO      |
 | shellybluht       | Shelly BLU H&T                                         | SBMO      |
 
+### Special thing types
```
Comment (jlaur) [inline bundles/org.openhab.binding.shelly/README.md:130]:
```suggestion
### Special Thing Types
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Two minor README comments.

File: bundles/org.openhab.binding.shelly/README.md (hunk 1 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/README.md:1570]: removed

File: bundles/org.openhab.binding.shelly/README.md (hunk 2 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/README.md:130]: changed

File: bundles/org.openhab.binding.shelly/README.md (hunk 1 above)
Comment (jlaur) [reply bundles/org.openhab.binding.shelly/README.md]: Sorry, I suggested also to correct a type, but have to admit it was well hidden.

Comment (markus7017) [reply bundles/org.openhab.binding.shelly/README.md]: :-)

**Comment by markus7017** [comment]
@jlaur why is there a problem with pom.xml so CI builds fail?

**Comment by jlaur** [comment]
> why is there a problem with pom.xml so CI builds fail?

This is related to the current build of 4.2.0, where pom files have been updated with new version number, but without applying spotless. #17016 will fix that. Don't worry about it, we can trigger rebuild after that PR is merged.

**Comment by markus7017** [comment]
@jlaur anything open for this PR?

File: bundles/org.openhab.binding.shelly/README.md [hunk 3]
```diff
@@ -1551,6 +1560,11 @@ See notes on discovery of Shelly BLU devices above.
 | battery | batteryLevel  | Number   | yes       | Battery Level in %                                      |
 |         | lowBattery    | Switch   | yes       | Low battery alert (< 20%)                               |
 
+## Shelly BLU Gateway (thing-type: shellyblugw)
+
+There are no additional channels beside the device group.
+
+
```
Comment (jlaur) [inline bundles/org.openhab.binding.shelly/README.md:1567]:
```suggestion
```

File: bundles/org.openhab.binding.shelly/README.md [hunk 4]
```diff
@@ -126,6 +127,14 @@ The binding provides the same feature set across all devices as good as possible
...
 | shellybluht       | Shelly BLU H&T                                         | SBMO      |
 
+### Special Thing Types
+
+| thing-type        | Model                                                  | Vendor ID |
+| ----------------- | ------------------------------------------------------ | --------- |
+| shellydevice      | A password protected Shelly device or an unknown type  |           |
+| shellyunknown     | An unknown Shelly device / model has been detected     |           |
+
+
```
Comment (jlaur) [inline bundles/org.openhab.binding.shelly/README.md:137]:
```suggestion
```

**Review by jlaur — APPROVED** [review]
LGTM now.

**Comment by markus7017** [comment]
LGTM

**Pull request merged**